src/Features/LanguageServer/Protocol/Features/Options/InternalDiagnosticsOptionsStorage.cs

```diff
@@ -8,21 +8,22 @@ namespace Microsoft.CodeAnalysis.Diagnostics;
 
 internal static class InternalDiagnosticsOptionsStorage
 {
-    private static readonly EditorConfigValueSerializer<DiagnosticMode> s_editorConfigValueSerializer = EditorConfigValueSerializer.CreateSerializerForEnum<DiagnosticMode>();
```

Comment:
the crux of this change is that we don't actually support any serializing of the options to control the underlyign diagnosticmode.  it's always going to be 'pull' for razor/liveshare.  And it's always going to be 'default' for normal diagnostics.  'default' means we then go look at the control tower feature flag (inside vs), which allows us to flip the option back to solution-crawler if we have to pull the emergency brake.